Scripts/DCS-BIOS/lib/modules/aircraft_modules/F-15E.lua

```diff
@@ -1171,4 +1171,590 @@ F_15E:defineMultipositionSwitch("R_LCP_LASER_DIG_4", 44, 3073, 73, 8, 0.1, "Rear
+PB%d%d 							PB01 to PB20 or PB10 to PB20
+[random prefix]_PB%d%d_LABEL 	PACS_PB01-20_LABEL
+PB%d%d-%d%d						PB01-02 for split on PB01 and PB02
+PB%d%d_PB%d%d					PB12_PB13 for split on PB12 and PB13
+[any prefix]_PB%d				TFR_PB5
+[any prefix]_PB%d%d				TFR_PB15 or TPOD_PB01 to TPOD_PB20 or SWPN_PB04
+PB%d%d_Value					is data begins 3 lines after "children are {" line
+PB%d%d%d%d						PB0405 for split between 04 and 05
+]]--
+
+local function parseMFDButtons(stringindication)
+	local input = split(stringindication, "%c")
+	local values = {};
+	for numline, checkline in pairs(input) do
+		if checkline  and string.sub(checkline, 1, 1) == "-" then
+			local start = numline+1
+			local val = input[start]
+			local double = false
+			local done = false
+			if val then
+				local pos = string.match(val, "^PB(%d[%d]?)$")
+				if not pos then
+					pos = string.match(val, "^[%a+_]+PB(%d+)[_LABEL]-$")
+					if not pos then
+						pos = string.match(val, "^PB(%d+)%-%d+$")
+						if not pos then
+							pos = string.match(val, "^PB(%d%d)[%d%d|_PB%d%d]+$")
+							double = true
+						else
+						  	double = true
+						end
+					end
+			  	end
+				if pos then
+					local num = tonumber(pos)
+					local combine = ""
+					for i = start+1, #input do
+						local line = input[i]
+						if line then
+							if string.sub(line, 1,1) ~="-" and line ~= "children are {" then
+								combine = combine .. line
+							elseif line == "children are {" then
+								combine = combine .. "$"
+								for j = i+3, #input do
+									local childline = input[j]
+									if line and string.sub(childline, 1, 1) ~= "}" then
+										combine = combine .. childline
+									else
+										break
+									end
+								end
+							else
+								break
+							end  
+						end
+					end
+					--print(num .. ", " .. val .. ": " .. combine)
+					if not double then
+						if not values[num] or string.match(values[num], "^[`%^]$") then
+							values[num] = combine
+						end
+					else
+						if (not values[num] and not values[num+1]) or (string.match(values[num], "^[`%^]$") and string.match(values[num+1], "^[`%^]$")) then
+							local startnum = string.find(combine, "%d")
+							values[num + (num>10 and 1 or 0)] = string.sub(combine, 1, startnum-1)
+							values[num + (num>10 and 0 or 1)] = string.sub(combine, startnum, #combine)
+						end
+					end
+				end  
+			end 
+		end
+	end
+
+	for i = 1, 20 do
+		if not values[i] then
+			values[i] = "          the button ind"
+		end
+	end
+	return values
+end
+
+F_15E:addExportHook(function()
+	local input = list_indication(3)
+	if not input then
+		return
+	end
+	F_MPDL_BUTTONS = parseMFDButtons(input)
+
+	input = list_indication(5)
+	if not input then
+		return
+	end
+	
+	F_MPCD_BUTTONS = parseMFDButtons(input)
+
+	input = list_indication(7)
+	if not input then
+		return
+	end
+	
+	F_MPDR_BUTTONS = parseMFDButtons(input)
+end)
+
+F_15E:defineString("F_MPDL_PBLABEL_01", function() 
+    return F_MPDL_BUTTONS[1] 
```

Comment:
Rather than 1-20, can we name these L1, L2, R1, R2, etc? I'm assuming that's the terminology that's used rather than telling someone "oh yeah that's button 16, do the math"

Reply:
Hm I don't love this, but it looks like that's also what we currently call them in bios so at least it's consistent with that. This is fine as-is.